vllm/v1/core/scheduler.py

```diff
@@ -290,6 +303,9 @@ def schedule(self) -> "SchedulerOutput":
             free_encoder_input_ids=self.encoder_cache_manager.get_freed_ids(),
         )
 
+        if self.log_stats:
```

Comment:
One thing that got us in trouble in V0 is too many loops over these types of data structures. 

I wonder if the creation of the request stats could be done in the list comprehensions used above,